rosidl_generator_py/cmake/rosidl_generator_py_get_typesupports.cmake

```diff
@@ -12,19 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-macro(accumulate_typesupports)
-  set(_typesupport_impl "")
-  if(${rmw_implementation}_FOUND)
-    get_rmw_typesupport(_typesupport_impl ${rmw_implementation} LANGUAGE "C")
-    list_append_unique(_typesupport_impls ${_typesupport_impl})
-  endif()
-endmacro()
-
 macro(rosidl_generator_py_get_typesupports TYPESUPPORT_IMPLS)
-  set(TYPESUPPORT_IMPLS "")
-  set(_typesupport_impls "")
-  call_for_each_rmw_implementation(accumulate_typesupports)
-  foreach(_typesupport_impl ${_typesupport_impls})
-    list_append_unique(TYPESUPPORT_IMPLS ${_typesupport_impl})
```

Comment:
The previous logic was broken. The variable which name is passed in `TYPESUPPORT_IMPLS` is never set by the macro. This only happened to work if the caller expected the result in `_typesupport_impls`. :exploding_head:

Reply:
Oh, `cmake` macros are really tricky.